### PR TITLE
language detection: Detect .NET processes

### PR DESCRIPTION
**Pull request by vitkyrka**
### What does this PR do?

Add detection of .NET process by using /proc/PID/maps to check if the process has mapped a standard .NET dll. This works for non-single-file deployments (both self-contained and framework-dependent), and framework-dependent single-file deployments.

It does not work for self-contained single-file deployments since these do not have any DLLs in their maps file.

Note that cross-platform binary deployments are already handled by the command-line based detector since they are executed with "dotnet foo.dll" or similar.

### Motivation

https://datadoghq.atlassian.net/browse/USMON-1163

### CODE REVIEW

File: pkg/languagedetection/privileged/privileged_detector_test.go [hunk 1]
```diff
@@ -149,3 +150,69 @@ func TestShortLivingProc(t *testing.T) {
...
+
+func TestDetectorAllFails(t *testing.T) {
+	MockPrivilegedDetectors(t, []languagemodels.Detector{DummyDetector{}})
+
+	d := NewLanguageDetector()
+	res := d.DetectWithPrivileges([]languagemodels.Process{DummyProcess{}})
+	require.Len(t, res, 1)
+	require.NotNil(t, res[0])
+	assert.Equal(t, languagemodels.Unknown, res[0].Name)
+}
```
Comment (Yumasi) [inline pkg/languagedetection/privileged/privileged_detector_test.go:218]: Nit: the checks are duplicated in all these tests, wouldn't it make sense to turn these into subtests with shared check logic?

**Review by Yumasi — APPROVED** [review]

File: pkg/languagedetection/privileged/privileged_detector_test.go (hunk 1 above)
Comment (vitkyrka) [reply pkg/languagedetection/privileged/privileged_detector_test.go]: Done.

**Pull request merged**